substrate/client/rpc-spec-v2/src/chain_head/chain_head.rs

```diff
@@ -86,12 +96,26 @@ const MAX_ONGOING_OPERATIONS: usize = 16;
 /// before paginations is required.
 const MAX_STORAGE_ITER_ITEMS: usize = 5;
 
+/// Suspend the subscriptions if the distance between the leaves and the current finalized
+/// block is larger than this value.
+///
+/// Subscriptions are suspended for the `suspended_duration`.
+const SUSPEND_ON_LAGGING_DISTANCE: usize = 128;
+
+/// The amount of time for which the subscriptions are suspended.
+///
+/// Subscriptions are suspended when the distance between any leaf
+/// and the finalized block is too large.
+const SUSPENDED_DURATION: Duration = Duration::from_secs(30);
```

Comment:
An alternative for this would be to simply recalculate the:

```rust
leaves = blockchain.leaves()?;
if leaf - finalized > 128 { return Err(); }
```

Reply:
Is there much of a computation cost for doing the above? Personally I think fewer constants is a good thing, but if it's expensive to compute then having a delay makes sense :)

Reply:
Yep that makes sense, I was also wondering if the `suspended duration` is overkill for this.

We might have a slight increase in reading the blockchain leaves numbers if they are placed on the disk, however I think we can live with that, since this is only an edge-case of an edge-case. 

Have removed that constant and we should allow subscriptions to resume normal work as soon as the distance is reasonable.